src/charm.py

```diff
@@ -4,24 +4,50 @@
 
 """Charmed Machine Operator for MySQL."""
 
+import hashlib
 import logging
+import secrets
+import string
 
 from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
 
-from mysqlsh_helpers import MySQL
+from mysqlsh_helpers import (
+    MySQL,
+    MySQLConfigureInstanceError,
+    MySQLConfigureMySQLUsersError,
+)
 
 logger = logging.getLogger(__name__)
 
+PASSWORD_LENGTH = 24
+PEER = "mysql-replicas"
+
+
+def generate_random_password(length: int) -> str:
```

Comment:
Wonder if we should put these two on the helpers

Reply:
I was thinking that `mysqlsh_helpers` will only contain helpers for mysql. these are functions at the moment, because i expect them to be replaced with `data-platform-libs` in the future